### PR TITLE
Refactor window state handling

### PR DESCRIPTION
**Pull request by Vultraz**
This PR refactors and simplifies the handling of window states in SDL2. It also improves behavior on changing states such as fullscreen/maximized/resolution and the saving of said states between sessions.

### CODE REVIEW

File: src/gui/dialogs/lobby_main.cpp [hunk 1]
```diff
@@ -439,26 +438,7 @@ static void signal_handler_sdl_video_resize(const event::tevent event,
...
-	// Setting to fullscreen doesn't seem to generate a resize event.
-	const SDL_Rect& rect = screen_area();
-
-	SDL_Event event;
-#if SDL_VERSION_ATLEAST(2, 0, 0)
-	event.type = SDL_WINDOWEVENT;
-	event.window.event = SDL_WINDOWEVENT_RESIZED;
-	event.window.data1 = rect.w;
-	event.window.data2 = rect.h;
-#else
```
Comment (aginor) [inline src/gui/dialogs/lobby_main.cpp]: I think we should retain this for SDL 1.2 for now

File: src/gui/dialogs/title_screen.cpp [hunk 2]
```diff
@@ -170,26 +169,7 @@ static void animate_logo(size_t& timer_id,
...
-	const SDL_Rect& rect = screen_area();
-
-	SDL_Event event;
-#if SDL_VERSION_ATLEAST(2, 0, 0)
-	event.type = SDL_WINDOWEVENT;
-	event.window.event = SDL_WINDOWEVENT_RESIZED;
-	event.window.data1 = rect.w;
-	event.window.data2 = rect.h;
-#else
-	event.type = SDL_VIDEORESIZE;
```
Comment (aginor) [inline src/gui/dialogs/title_screen.cpp]: I think we should retain this for SDL 1.2 for now.

**Comment by gfgtdf** [comment]
Hmm there are a lot of commits in this pr, which makes this rather hard to review, what i would recomend is:

1) There are some very simple commits, specially
https://github.com/Vultraz/wesnoth/commit/65a756c3c777f3f4e8184a638fb07a643b983da7
https://github.com/Vultraz/wesnoth/commit/f10dd53013aad2041b7d4ab81052378fad199160
which i would commit directly to master instead of having them in this pr.

2) There are some fixup commits for your commits, for example
https://github.com/Vultraz/wesnoth/commit/e311a5c286d5646082327b3aa534d7be486ea89c
I reccomend to merge them directly into the commits that made them nessecary, sometimes you want to have those in seperate commits but in this case they don't really have useful information in their commit messages or similar, so i dont really see a reason to have those in seperate commits.

3) There are some commits that delete code without further explanation, specially
https://github.com/Vultraz/wesnoth/commit/da6346d6c9538e82edc71f1ef6b02d9bd67bcc08
You should add explanation why you removed that code. Ideally something like 'this is no needed anymore becasue sine commit xxxx this work is now done by yyyy'. Or something like 'This code was a fix for bug #12345 but it was tested that it wasn't needed (bug doesnt happen anymore even with that code removed)'

4) There are some commits that change how bbp settings are handled, i dotn really understand what it does but i think it'd be reasier to review it it was in seperate prs.

File: src/gui/dialogs/lobby_main.cpp (hunk 1 above)
Comment (Vultraz) [inline src/gui/dialogs/lobby_main.cpp]: Fixed and re-pushed.

**Comment by Vultraz** [comment]
Some of these commits have been moved (with edits) to PR #568. I'll update this PR once that is merged.

**Comment by Vultraz** [comment]
Rebased and updated.

**Pull request merged**